crates/bevy_asset/src/handle.rs

```diff
@@ -373,34 +380,32 @@ impl HandleUntyped {
     /// If this handle is strong and dropped, there is no guarantee that the asset
     /// will still be available (if only the returned handle is kept)
     pub fn typed_weak<T: Asset>(&self) -> Handle<T> {
-        self.clone_weak().typed()
+        self.clone_weak().typed().unwrap()
     }
 
     /// Converts this handle into a typed [`Handle`] of an [`Asset`] `T`.
     ///
     /// The new handle will maintain the Strong or Weak status of the current handle.
-    ///
-    /// # Panics
-    ///
-    /// Will panic if type `T` doesn't match this handle's actual asset type.
-    pub fn typed<T: Asset>(mut self) -> Handle<T> {
+    pub fn typed<T: Asset>(mut self) -> Result<Handle<T>, HandleError> {
         if let HandleId::Id(type_uuid, _) = self.id {
```

Comment:
Weird question, but what happens if this `if` fails?

Reply:
Yeah, that's a bit suspicious 🤔

Reply:
nothing
the assertion is just possible on id handles, not on asset path handles